internal/controller/api/v1alpha1/cluster_types.go

```diff
@@ -25,17 +25,86 @@ import (
 
 // ClusterSpec defines the desired state of Cluster
 type ClusterSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Cluster name
+	Name string `json:"name"`
+	// Installation will start once cluster is in known state and approved
+	Approved                 bool   `json:"approved,omitempty"`
+	OpenshiftVersion         string `json:"openshift_version"`
+	BaseDnsDomain            string `json:"base_dns_domain,omitempty"`
+	ClusterNetworkCidr       string `json:"cluster_network_cidr,omitempty"`
+	ClusterNetworkHostPrefix int64  `json:"cluster_network_host_prefix,omitempty"`
+	ServiceNetworkCidr       string `json:"service_network_cidr,omitempty"`
+	ApiVip                   string `json:"api_vip,omitempty"`
+	ApiVipDnsName            string `json:"api_vip_dns_name,omitempty"`
+	IngressVip               string `json:"ingres_vip,omitempty"`
+	MachineNetworkCidr       string `json:"machine_network_cidr,omitempty"`
+	SshPublicKey             string `json:"ssh_public_key,omitempty"`
+	VipDhcpAllocation        bool   `json:"vip_dhcp_allocation,omitempty"`
+	HttpProxy                string `json:"http_proxy,omitempty"`
+	HttpsProxy               string `json:"https_proxy,omitempty"`
+	NoProxy                  string `json:"no_proxy,omitempty"`
+	UserManagedNetworking    bool   `json:"user_managed_networking,omitempty"`
+	AdditionalNtpSource      string `json:"additional_ntp_source,omitempty"`
+	InstallConfigOverrides   string `json:"install_config_overrides,omitempty"`
+	// The name of the secret containing the pull secret
+	PullSecretName string `json:"pull_secret_name"`
+}
+
+// +kubebuilder:validation:Enum=insufficient;ready;error;preparing-for-installation;pending-for-input;installing;finalizing;installed;adding-hosts;cancelled;installing-pending-user-action
+type ClusterState string
```

Comment:
Use string alias rather than enum, as here:
https://github.com/kubernetes/api/blob/fcac651617f2c6b75a129dc2dcbdfd3aa8b0aa1c/core/v1/types.go#L4816

Reply:
In that case i can probably just make it a string without enum, because users cannot affect this filed, and then i will not need to change two places when changing state machine in swagger. 
Using enum update the yaml to have all the options, like a documentation this is why i used enum.